CommonCPack.cmake

```diff
@@ -1,225 +1,276 @@
-  set(CPACK_COMPONENT_DOC_DEPENDS lib)
+# Setup all CPACK variables
 
-  set(CPACK_COMPONENT_APPS_DISPLAY_NAME "${CPACK_PROJECT_NAME} Applications")
-  set(CPACK_COMPONENT_APPS_DESCRIPTION "${CPACK_PROJECT_NAME} Applications")
-  set(CPACK_COMPONENT_APPS_DEPENDS lib)
+common_cpack_set_base_parameters()
 
-  set(CPACK_COMPONENT_EXAMPLES_DISPLAY_NAME "${CPACK_PROJECT_NAME} Examples")
-  set(CPACK_COMPONENT_EXAMPLES_DESCRIPTION
-    "${CPACK_PROJECT_NAME} Example Source Code")
-  set(CPACK_COMPONENT_EXAMPLES_DEPENDS dev)
-elseif(CPACK_COMPONENTS_ALL STREQUAL "none")
-  set(CPACK_COMPONENTS_ALL)
+if(NOT CPACK_COMPONENTS_ALL)
+  common_cpack_set_components_all(${PROJECT_NAME})
 endif()
 
-include(LSBInfo)
-
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  find_program(RPM_EXE rpmbuild)
-  find_program(DEB_EXE debuild)
-endif()
+  # Legacy?
+  if(EXISTS ${PROJECT_SOURCE_DIR}/CMake/${PROJECT_NAME}.in.spec)
+    configure_file(${PROJECT_SOURCE_DIR}/CMake/${PROJECT_NAME}.in.spec
+      ${PROJECT_SOURCE_DIR}/CMake/${PROJECT_NAME}.spec @ONLY)
+  endif()
 
-# Auto-package-version magic
-include(GitInfo)
-set(CMAKE_PACKAGE_VERSION "" CACHE
-  STRING "Additional build version for packages")
-mark_as_advanced(CMAKE_PACKAGE_VERSION)
-
-if(GIT_REVISION)
-  if(NOT PACKAGE_VERSION_REVISION STREQUAL GIT_REVISION)
-    if(PACKAGE_VERSION_REVISION)
-      if(CMAKE_PACKAGE_VERSION)
-        math(EXPR CMAKE_PACKAGE_VERSION "${CMAKE_PACKAGE_VERSION} + 1")
-      else()
-        set(CMAKE_PACKAGE_VERSION "1")
-      endif()
-    else()
-      set(CMAKE_PACKAGE_VERSION "")
-    endif()
-    set(CMAKE_PACKAGE_VERSION ${CMAKE_PACKAGE_VERSION} CACHE STRING
-      "Additional build version for packages" FORCE)
+  # Optional: append ABI version to package name and make a list of old packages
+  if(${UPPER_PROJECT_NAME}_PACKAGE_USE_ABI AND ${PROJECT_NAME}_VERSION_ABI)
+    set(_abi_version ${${PROJECT_NAME}_VERSION_ABI})
+    get_old_package_names(${CPACK_PACKAGE_NAME} ${_abi_version} _old_packages)
+    set(CPACK_PACKAGE_NAME "${CPACK_PACKAGE_NAME}${_abi_version}")
   endif()
-  set(PACKAGE_VERSION_REVISION ${GIT_REVISION} CACHE INTERNAL "" FORCE)
-endif()
 
-# Heuristics to figure out cpack generator
-if(MSVC)
-  set(CPACK_GENERATOR "NSIS")
-  set(CPACK_NSIS_MODIFY_PATH ON)
-elseif(APPLE)
-  set(CPACK_GENERATOR "PackageMaker")
-  set(CPACK_OSX_PACKAGE_VERSION "${${UPPER_PROJECT_NAME}_OSX_VERSION}")
-elseif(LSB_DISTRIBUTOR_ID MATCHES "Ubuntu")
-  set(CPACK_GENERATOR "DEB")
-elseif(LSB_DISTRIBUTOR_ID MATCHES "RedHatEnterpriseServer")
-  set(CPACK_GENERATOR "RPM")
-elseif(DEB_EXE)
-  set(CPACK_GENERATOR "DEB")
-elseif(RPM_EXE)
-  set(CPACK_GENERATOR "RPM")
-else()
-  set(CPACK_GENERATOR "TGZ")
+  # Convert PACKAGE_[REPLACES|CONFLICTS] list to string format without duplicate
+  set(_package_replaces ${${UPPER_PROJECT_NAME}_PACKAGE_REPLACES})
+  list(APPEND _package_replaces ${_old_packages}) # TODO is this really needed?
```

Comment:
It's hard to say, I don't think regular debian ubuntu packages do that, but they also follow more rules (dev and lib are split, etc). I could try to make some tests to see if dpkg accepts to install libdeflect7 on top of libdeflect6 without this to see if it makes a difference...